src/mixins/filing-template-mixin.ts

```diff
@@ -291,6 +277,22 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
       })
     }
 
+    // restore the Amalgamating Businesses
+    if (draftFiling.amalgamationApplication.amalgamatingBusinesses) {
+      this.setAmalgamatingBusinesses([
+        ...draftFiling.amalgamationApplication.amalgamatingBusinesses
+      ])
+
+      // Re-fetch the business table data. We do this in case one of the businesses has changed
+      // (eg, became historical) since the last draft save.
+      const holdingPrimary = await this.refetchAmalgamatingBusinessesInfo()
+
+      // If there's a holding or primary business, fetch its data and update the prepopulated data.
+      // This will overwrite office addresses, directors, share structure, legal name and legal type
+      // that were set above from draft data.
+      if (holdingPrimary) await this.updatePrepopulatedData(holdingPrimary)
+    }
```

Comment:
This is called when a draft amalgamation is restored. This is to catch the scenario where the holding/primary business name or type has changed in the background and then this amalgamation is resumed.

This block of code was moved down (in particular, below the name request code) as it may overwrite the legal name and legal type set above.